From 7a1d6fec02616c1eca0e6f901c1b76c2a4d285fe Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@apache.org>
Date: Wed, 28 Jan 2015 23:58:34 +0000
Subject: [PATCH 0005/1164] HIVE-9436 : RetryingMetaStoreClient does not retry
 JDOExceptions (Sushanth Sowmyan, reviewed by
 Thejas M Nair)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1655501 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit a1715dd657c4cd6198cc0d3665ebb81812f44a07)
---
 .../hive/metastore/RetryingMetaStoreClient.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java b/metastore/src/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java
index 4c9299c..b4f02fc 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java
@@ -98,7 +98,7 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
             (e.getCause() instanceof TTransportException)) {
           caughtException = (TException) e.getCause();
         } else if ((e.getCause() instanceof MetaException) &&
-            e.getCause().getMessage().matches("JDO[a-zA-Z]*Exception")) {
+            e.getCause().getMessage().matches("(?s).*JDO[a-zA-Z]*Exception.*")) {
           caughtException = (MetaException) e.getCause();
         } else {
           throw e.getCause();
-- 
1.7.9.5

